src/python/aqueduct_executor/operators/function_executor/install_requirements.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+import base64
+import subprocess
+import sys
+
+from aqueduct_executor.operators.function_executor.spec import FunctionSpec, parse_spec
+from aqueduct_executor.operators.utils import utils
+from aqueduct_executor.operators.utils.enums import FailureType
+from aqueduct_executor.operators.utils.execution import ExecFailureException, ExecutionState, Logs
+from aqueduct_executor.operators.utils.storage.parse import parse_storage
+
+
+def install_missing_packages(missing_path: str, spec: FunctionSpec) -> None:
+    install_output = subprocess.run(
+        [sys.executable, "-m", "pip", "install", "-r", missing_path], capture_output=True, text=True
+    )
+
+    if install_output.returncode != 0:
+        exception = ExecFailureException(
+            failure_type=FailureType.USER_FATAL,
+            tip="We are unable to install certain dependency packages. Please remove them from the requirement file and try again.",
+        )
+        from_exception_exec_state = ExecutionState.from_exception(
+            exception, user_logs=Logs(stdout=install_output.stdout, stderr=install_output.stderr)
+        )
+
+        utils.write_exec_state(
+            parse_storage(spec.storage_config), spec.metadata_path, from_exception_exec_state
+        )
+        sys.exit(1)
+
+
+def run(local_path: str, requirements_path: str, missing_path: str, spec: FunctionSpec) -> None:
+    with open(local_path, "r") as f:
+        local_req = set(f.read().split("\n"))
+
+    with open(requirements_path, "r") as f:
+        required = f.read().split("\n")
+
+    missing = []
+    for r in required:
+        # Remove any @ file because we may not have those files local to the user's device in our file system.
+        if r not in local_req and "@ file" not in r:
+            missing.append(r)
+
+    if len(missing) > 0:
```

Comment:
I think we should use an if-else to handle `len(missing) == 0` in `main()`, otherwise the path will be empty and `run()` would fail.

Reply:
I think in the start script we are always inputting the `missing_path` argument, so the path will never be empty? Also the missing dependency installation happens only when `len(missing) > 0`, so it seems that it won't attempt to touch that missing file when there's no missing dependency?

Reply:
The issue is if `len(missing) > 0` the file is not touched it will not be created at all. This line is fine, but on 14 it will error since the path is never created. We should check if the path exists or if `len(missing) > 0` before call `install_missing_packages()`.

Reply:
Oh actually I realized you call `install_missing_packages()` under this block, so everything should work!